### ISSUE TITLE
button href is not updated when component is rerendered

### ISSUE THREAD

**Issue by bvella** [facebook/react#6111]
Using react 0.14.7.

It seems that for some reason, when react diffs the dom,
- for buttons it replaces the text, but is not updating the href attribute
- for anchors it is correctly updating both the text and the href attribute

The below code illustrates the issue. It is a component that cycles its internal state and renders a link and a button to move the state to the next stage. the link works fine but the button does not as the href is not updated, but the text is.

```
var Bug = React.createClass({

    getInitialState: function() {
        return {
            path: ""
        };
    },

    render: function () {
        if (this.state.path === '') {
            return <div>STAGE 1 ::
                    <a href="1" onClick={this._changeLocation}>Click to go to 2</a>
                    <button href="1" onClick={this._changeLocation}>Click to go to 2</button>
                </div>;
        } else if (this.state.path === '1') {
            return <div>STAGE 2 ::
                    <a href="2" onClick={this._changeLocation}>Click to go to 3</a>
                    <button href="2" onClick={this._changeLocation}>Click to go to 3</button>
                </div>;
        } else {
            return <div>STAGE 3 ::
                    <a href="" onClick={this._changeLocation}>Start Over</a>
                    <button href="" onClick={this._changeLocation}>Start Over</button>
                </div>;
        }
    },

    _changeLocation: function (e) {
        e.preventDefault();
        this.setState({path: e.currentTarget.getAttribute("href")});
    }

});
```

**Comment by jimfb** [comment]
Appears to work for me on latest/future: http://jsfiddle.net/wwyabmyx/

Does the fiddle above have your desired behavior?

**Comment by bvella** [comment]
yes, indeed. It seems to be fixed in 0.15.

here's a fiddle with 0.14.6 that show the above code not working with the button:  https://jsfiddle.net/wbrmz838/

**Comment by jimfb** [comment]
Ok, I'm going to close this out as fixed in master.

**Comment by bvella** [comment]
Is there any workaround or backport on 0.14? or when can we expect 0.15 to be stable?

**Comment by zpao** [comment]
The issue is that `href` isn't actually a supported property on `button`s, it doesn't mean anything. So our support for it was focused on the valid case of anchor tags. So we set `node.href`. For anchors the property is linked to the attribute but that's not what happens with buttons. v15 changed the default so we'll use the attribute for more things than we did before. 15 will be stable soon.

Ultimately though this is a case I would generally argue should use `data-*` attributes if you really need to read data from the DOM. Even better would be to write the code so you don't have to write and read from DOM nodes and instead manage some state in the component.
